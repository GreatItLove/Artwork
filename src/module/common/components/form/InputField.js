import React from 'react';
import TextField from 'material-ui/TextField';
import PropTypes from 'prop-types';

const propTypes = {
  meta: PropTypes.object.isRequired,
  input: PropTypes.object.isRequired
};

function InputField({ meta: { touched, error } = {}, input: { ...inputProps }, ...props }) {
  return <TextField errorText={touched && error} {...inputProps} {...props} fullWidth />;
}

InputField.propTypes = propTypes;

export default InputField;
