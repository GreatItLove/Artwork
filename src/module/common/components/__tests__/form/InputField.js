import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import InputField from '../../form/InputField';

describe('Input Field', () => {
  it('Snapshot of Input Field', () => {
    const meta = {};
    const input = {};
    const wrapper = shallow(<InputField {...{ meta, input }} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
