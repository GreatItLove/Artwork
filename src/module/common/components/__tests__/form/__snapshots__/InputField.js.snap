// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Field Snapshot of Input Field 1`] = `
<TextField
  disabled={false}
  floatingLabelFixed={false}
  fullWidth={true}
  multiLine={false}
  rows={1}
  type="text"
  underlineShow={true}
/>
`;
